contracts/strategies/QVSimple.sol

```diff
@@ -0,0 +1,270 @@
+    /// ========== Modifier ============
+    /// ================================
+
+    /// @notice Modifier to check if the allocation has ended
+    /// @dev Reverts if the allocation has not ended
+    modifier onlyAfterAllocation() {
+        _checkOnlyAfterAllocation();
+        _;
+    }
+
+    /// ====================================
+    /// ==== External/Public Functions =====
+    /// ====================================
+
+    /// @notice Add allocator
+    /// @dev Only the pool manager(s) can call this function and emits an `AllocatorAdded` event
+    /// @param _allocator The allocator address
+    function addAllocator(address _allocator) external onlyPoolManager(msg.sender) {
+        allowedAllocators[_allocator] = true;
+
+        emit AllocatorAdded(_allocator, msg.sender);
+    }
+
+    /// @notice Remove allocator
+    /// @dev Only the pool manager(s) can call this function and emits an `AllocatorRemoved` event
+    /// @param _allocator The allocator address
+    function removeAllocator(address _allocator) external onlyPoolManager(msg.sender) {
+        allowedAllocators[_allocator] = false;
+
+        emit AllocatorRemoved(_allocator, msg.sender);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Check if the allocation has ended
+    /// @dev Reverts if the allocation has not ended
+    function _checkOnlyAfterAllocation() internal view virtual {
+        if (block.timestamp <= allocationEndTime) revert ALLOCATION_NOT_ENDED();
+    }
+
+    /// @notice Distribute the tokens to the recipients
+    /// @dev The '_sender' must be a pool manager and the allocation must have ended
+    /// @param _recipientIds The recipient ids
+    /// @param _sender The sender of the transaction
+    function _distribute(address[] memory _recipientIds, bytes memory, address _sender)
+        internal
+        virtual
+        override
+        onlyPoolManager(_sender)
+        onlyAfterAllocation
+    {
+        uint256[] memory payouts = _votingState._getPayout(_recipientIds, poolAmount);
+
+        for (uint256 i; i < payouts.length;) {
+            address recipientId = _recipientIds[i];
+            Recipient memory recipient = _recipients[recipientId];
+
+            uint256 amount = payouts[i];
+
+            if (paidOut[recipientId] || !_isAcceptedRecipient(recipientId) || amount == 0) {
+                revert RECIPIENT_ERROR(recipientId);
+            }
+
+            IAllo.Pool memory pool = allo.getPool(poolId);
+            _transferAmount(pool.token, recipient.recipientAddress, amount);
+
+            paidOut[recipientId] = true;
+
+            emit Distributed(recipientId, abi.encode(recipient.recipientAddress, amount, _sender));
+            unchecked {
+                ++i;
+            }
+        }
+        if (!distributionStarted) {
+            distributionStarted = true;
+        }
```

Comment:
I think this should be moved to the beginning of the function to prevent reentrancy issues